From 73f077204f70a157c3e1b3c63996ac29b005ea5b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?G=C3=BCnther=20Deschner?= <gd@samba.org>
Date: Mon, 15 Jan 2018 23:20:39 +0100
Subject: [PATCH] FIXME: build: fix libceph-common detection

Guenther

Signed-off-by: Guenther Deschner <gd@samba.org>
---
 source3/wscript | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/source3/wscript b/source3/wscript
index 6d2d94bae87..48a9271f002 100644
--- a/source3/wscript
+++ b/source3/wscript
@@ -1547,11 +1547,13 @@ main() {
         conf.env['CPPPATH_CEPHFS'] = Options.options.libcephfs_dir + '/include'
         conf.env['LIBPATH_CEPHFS'] = Options.options.libcephfs_dir + '/lib'
         conf.env['LIBPATH_CEPH-COMMON'] = Options.options.libcephfs_dir + '/lib/ceph'
+    else:
+        conf.env['LIBPATH_CEPH-COMMON'] = Options.options.LIBDIR + '/ceph'
 
     if (Options.options.with_cephfs and
         conf.CHECK_HEADERS('cephfs/libcephfs.h', False, False, 'cephfs') and
-        conf.CHECK_LIB('cephfs', shlib=True) and
-        conf.CHECK_LIB('ceph-common', shlib=True)):
+        conf.CHECK_LIB('cephfs', shlib=True)):
+        conf.CHECK_LIB('ceph-common', shlib=True)
         if Options.options.with_acl_support:
             conf.DEFINE('HAVE_CEPH', '1')
             if conf.CHECK_FUNCS_IN('ceph_statx', 'cephfs ceph-common',
-- 
2.14.3

